share/spack/spack-completion.bash

```diff
@@ -363,9 +363,9 @@ function _spack_find {
     if $list_options
     then
         compgen -W "-h --help -s --short -p --paths -d --deps -l --long
-                    -L --very-long -f --show-flags -e --explicit
-                    -E --implicit -u --unknown -m --missing -v --variants
-                    -M --only-missing -N --namespace" -- "$cur"
+                    -L --very-long -f --show-flags -F --show-full-compiler
```

Comment:
If you think the option name should remain ```--show-compiler-version``` then this should change. That being said IMO if ```--show-compiler-version``` is going to show the flags and version I'd rather it was called ```--show-full-compiler``` anyway, so I'd prefer just changing the option name back in ```find```.